### PR TITLE
feat(agent-nodejs): generate typescript mongoose models

### PR DESCRIPTION
**Pull request by SteveBunlon**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: src/services/dumpers/templates/agent-nodejs/javascript/models/model.hbs [hunk 1]
```diff
@@ -2,7 +2,7 @@ const Mongoose = require('mongoose');
 
 const schema = new Mongoose.Schema({
 {{#each fields as |field|}}
-  {{wq field.name}}: {{#if field.ref}}{ type: {{field.type}}, ref: '{{field.ref}}' }{{else if (isObject field.type)}}{{>renderNested type=field.type level=1}}{{else}}{{field.type}}{{/if}},
+  {{wsc field.name}}: {{#if field.ref}}{ type: {{field.type}}, ref: '{{field.ref}}' }{{else if (isObject field.type)}}{{>renderNested type=field.type level=1}}{{else}}{{field.type}}{{/if}},
```
Comment (SteveBunlon) [inline src/services/dumpers/templates/agent-nodejs/javascript/models/model.hbs:5]: This needs to be taken out from this PR, and done using a refactor PR. 

This removes the quote for the attribute names when it is not needed

File: src/services/dumpers/templates/forest-express/models/mongo-model.hbs [hunk 2]
```diff
@@ -6,7 +6,7 @@ module.exports = (mongoose, Mongoose) => {
   // Learn more here: https://docs.forestadmin.com/documentation/reference-guide/models/enrich-your-models#declaring-a-new-field-in-a-model
   const schema = Mongoose.Schema({
   {{#each fields as |field|}}
-    {{wq field.name}}: {{#if field.ref}}{ type: {{field.type}}, ref: '{{field.ref}}' }{{else if (isObject field.type)}}{{>renderNested type=field.type level=2}}{{else}}{{field.type}}{{/if}},
+    {{wsc field.name}}: {{#if field.ref}}{ type: {{field.type}}, ref: '{{field.ref}}' }{{else if (isObject field.type)}}{{>renderNested type=field.type level=2}}{{else}}{{field.type}}{{/if}},
```
Comment (SteveBunlon) [inline src/services/dumpers/templates/forest-express/models/mongo-model.hbs:9]: This needs to be taken out from this PR in a fix PR. 

This prevents attributes without spaces or dashes to be quoted. Today, any of the attributes are being quoted, which is ugly.

File: test/services/dumpers/agent-nodejs/agent-nodejs-dumper-mongoose-models.test.ts [hunk 3]
```diff
@@ -41,7 +41,7 @@ describe('services > dumpers > agentNodejsDumper > mongoose models', () => {
     await dumper.dump(config, schema);
   }
 
-  describe.each([languages.Javascript])('language: $name', language => {
+  describe.each([languages.Typescript])('language: $name', language => {
```
Comment (DrRaider) [inline test/services/dumpers/agent-nodejs/agent-nodejs-dumper-mongoose-models.test.ts:44]:
```suggestion
  describe.each([languages.Javascript, languages.Typescript])('language: $name', language => {
```

File: test/services/dumpers/agent-nodejs/agent-nodejs-dumper-mongoose-models.test.ts [hunk 4]
```diff
@@ -153,4 +137,41 @@ describe('services > dumpers > agentNodejsDumper > mongoose models', () => {
...
+
+        const generatedFile = fs.readFileSync(
+          `${appRoot}/test-output/${language.name}/mongodb/models/films.${language.fileExtension}`,
+          'utf8',
+        );
+
+        expect(generatedFile).toContain(`export { FilmsInterface, filmsSchema }`);
+      });
+    });
+  });
```
Comment (DrRaider) [inline test/services/dumpers/agent-nodejs/agent-nodejs-dumper-mongoose-models.test.ts:176]: I guess those can't be included in the loop right ?

File: test/services/dumpers/agent-nodejs/agent-nodejs-dumper-mongoose-models.test.ts (hunk 3 above)
Comment (SteveBunlon) [reply test/services/dumpers/agent-nodejs/agent-nodejs-dumper-mongoose-models.test.ts:44]: 😱 that's a nice catch

File: test/services/dumpers/agent-nodejs/agent-nodejs-dumper-mongoose-models.test.ts (hunk 4 above)
Comment (SteveBunlon) [reply test/services/dumpers/agent-nodejs/agent-nodejs-dumper-mongoose-models.test.ts:176]: Actually, in the future we could change the way index.js file is generated, to make it close to the index.ts one, but for now it is out of scope

File: src/services/dumpers/agent-nodejs.ts [hunk 5]
```diff
@@ -276,12 +279,39 @@ export default class AgentNodeJs extends AbstractDumper {
...
         fields: fieldsDefinition,
         timestamps: options.timestamps,
-      });
+        modelFileName,
+        modelPath,
+      };
+    });
+  }
+
+  private async writeMongooseModels(language: Language, schema) {
```
[the remark this answers is not in the document]
Comment (SteveBunlon) [reply src/services/dumpers/agent-nodejs.ts:293]: done

**Pull request merged**